addons/website/models/mixins.py

```diff
@@ -112,6 +112,23 @@ def _default_cover_properties(self):
             "resize_class": "o_half_screen_height",
         }
 
+    def write(self, vals):
+        result = True
+        for item in self:
+            copy_vals = dict(vals)
+            # Prevent updates on list displays from destroying resize_class.
+            if 'cover_properties' in copy_vals:
+                cover_properties = json_scriptsafe.loads(copy_vals['cover_properties'])
```

Comment:
@bso-odoo You also loads the JSON here at each loop step and check the resize class each time :grimacing: